### ISSUE TITLE
allow specifying volume options in kube's hostpath

### ISSUE THREAD

**Issue by ibotty** [containers/podman#20467]
### Feature request description

I want to mount a host volume (or better, another named volume) with `:O` to allow a CoW-Volume for a test container.

I can run a regular container with `-v` to do that, but no kube pod.

### Suggest potential solution

Either support named volumes that accept the same type as `-v`, (akin to `hostpath` `PersistentVolumeClaims` in kubernetes), or support a way to specify these volume flags in kube YAMLs via annotations.

### Have you considered any alternatives?

-

### Additional context

-

**Comment by rhatdan** [comment]
We don't like to modify the kube yaml, since our goal is to have compatibility between Kubernetes and Podman.   We have worked around this in the past by using Annotations.

**Comment by ibotty** [comment]
I don't propose modifying the kube yaml, but maybe accepting the hostpath's `path` to be a string as in podman's `-v`.  I do think the best option would be to allow these kinds of volumes as named volumes though.

**Comment by rhatdan** [comment]
Please give me an example of what you want, podman kube play command along with kube yaml.

**Comment by ibotty** [comment]
So we don't speak past each other: syntax is not what's really important for me.  I'd like to have the flexibility to use all the mount options I have for `podman run -v ...` also with kubernetes pods.

To propose something: With more flexible named volumes and a podman interface like
```
podman volume create myvol --overlay-dir /path/to/dir --with-persistent-workdir --with-persistent-lowerdir
```
which would correspond to `-v /path/to/dir:/container/dir:O,upperdir=/some/podman/internal/path,workdir=/some/other/podman/internal/path`

Maybe also support the other options, e.g.
```
podman volume create myvol --mount-host /some/path --readonly ```

**Comment by ibotty** [comment]
Anything else I might have to clarify?

**Comment by rhatdan** [comment]
@mheon @umohnani8 Thoughts?

**Comment by mheon** [comment]
@umohnani8 Don't we store these in annotations right now?

**Comment by umohnani8** [comment]
Yes, we do support this with annotations right now. Here is an example:
```
✗ podman create -v $PWD/bin:/myvol:Z fedora                                                                    
4453e2fc47c83d5c9a76228d8de7793f41ff0a03b28d7972e074f49e0b0feefa

✗ podman kube generate 4453e2fc47c8 
# Save the output of this file and use kubectl create -f to import
# it into Kubernetes.
#
# Created with podman-4.8.0-dev

# NOTE: If you generated this yaml from an unprivileged and rootless podman container on an SELinux
# enabled system, check the podman generate kube man page for steps to follow to ensure that your pod/container
# has the right permissions to access the volumes added.
---
apiVersion: v1
kind: Pod
metadata:
  annotations:
    bind-mount-options: /home/umohnani/go/src/github.com/containers/podman/bin:Z
  creationTimestamp: "2023-10-31T16:23:21Z"
  labels:
    app: stupefiedleavitt-pod
  name: stupefiedleavitt-pod
spec:
  containers:
  - image: registry.fedoraproject.org/fedora:latest
    name: stupefiedleavitt
    volumeMounts:
    - mountPath: /myvol
      name: home-umohnani-go-src-github.com-containers-podman-bin-host-0
  volumes:
  - hostPath:
      path: /home/umohnani/go/src/github.com/containers/podman/bin
      type: Directory
    name: home-umohnani-go-src-github.com-containers-podman-bin-host-0
```

We create a host path volume with generate kube and a bind mount annotation is added to the yaml as well so that when we do `podman kube play` we can use the bind mount options from the annotations when setting up the volumes.

@ibotty does this work for your use case?

**Comment by ibotty** [comment]
Sorry for taking so long.  I'll have to try it for real first, but this looks like it will work.  Is this documented somewhere?

**Comment by umohnani8** [comment]
Looks like we don't have this documented in the man pages, probably something worth fixing. But here is the PR that added this functionality that may help while you test it out https://github.com/containers/podman/pull/11812.

**Comment by rhatdan** [comment]
Would either @ibotty or @umohnani8 care to open a PR to document this in podman kube play man page?

**Comment by rhatdan** [comment]
@umohnani8 any update?
